_stbt/core.py

```diff
@@ -2055,156 +1952,147 @@ def _confirm_match(image, position, template, match_parameters):
         thresholded,
         cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (3, 3)),
         iterations=match_parameters.erode_passes)
-    log(absdiff, "confirm-absdiff")
-    log(thresholded, "confirm-absdiff_threshold")
-    log(eroded, "confirm-absdiff_threshold_erode")
+    imglog.add("confirm-absdiff", absdiff)
+    imglog.add("confirm-absdiff_threshold", thresholded)
+    imglog.add("confirm-absdiff_threshold_erode", eroded)
 
     return cv2.countNonZero(eroded) == 0
 
 
-_frame_number = 0
-
-
-def _log_image(image, name, directory):
-    if logging.get_debug_level() <= 1:
-        return
-    global _frame_number
-    if name == "source":
-        _frame_number += 1
-    d = os.path.join(directory, "%05d" % _frame_number)
-    try:
-        utils.mkdir_p(d)
-    except OSError:
-        warn("Failed to create directory '%s'; won't save debug images." % d)
-        return
-    with _numpy_from_sample(image, readonly=True) as img:
-        if img.dtype == numpy.float32:
-            img = cv2.convertScaleAbs(img, alpha=255)
-        cv2.imwrite(os.path.join(d, name) + ".png", img)
-
+class MatchImageLogger(logging.ImageLogger):
```

Comment:
Inheritance doesn't seem appropriate here.  `add` is overridden and `write_images` isn't used by `_match`.